pandas/core/index.py

```diff
@@ -4395,11 +4420,21 @@ def _reference_duplicate_name(self, name):
         # count the times name equals an element in self.names.
         return sum(name == n for n in self.names) > 1
 
-    def _format_native_types(self, **kwargs):
-        # we go through the levels and format them
-        levels = [level._format_native_types(**kwargs)
-                  for level in self.levels]
-        mi = MultiIndex(levels=levels, labels=self.labels, names=self.names,
+    def _format_native_types(self, na_rep='nan', **kwargs):
+        new_levels = []
+        new_labels = []
+        # go through the levels and format them
+        for level, label in zip(self.levels, self.labels):
+            level = level._format_native_types(na_rep=na_rep, **kwargs)
+            # add nan values
+            nan_index = len(level)
+            level = np.append(level, na_rep)
+            # add the corresponding value to the labels
+            label = [nan_index if _ < 0 else _ for _ in label]
+            new_levels.append(level)
+            new_labels.append(label)
+        # reconstruct the multi-index
+        mi = MultiIndex(levels=new_levels, labels=new_labels, names=self.names,
                         sortorder=self.sortorder, verify_integrity=False)
```

Comment:
But we return the `.values` of the multi-index... so even if there are extra levels (for instance if we add `na_rep` while there are no NaNs), they don't show up in `mi.values`, so I'm not quite sure of how to test that.

```
In [16]: mi
Out[16]: 
MultiIndex(levels=[[u'0', u'1', u'_'], [u'2', u'3', u'_']],
           labels=[[0, 1], [0, 1]],
           names=[u'a', u'b'])

In [17]: mi.values
Out[17]: array([('0', '2'), ('1', '3')], dtype=object)
```

But I agree, the code would be better if the presence of NaNs was checked. I'll add that.